mitodl/mit-open-login-button
# add basic code to generate the login button

### What are the relevant tickets?
Closes https://github.com/mitodl/mit-open-login-button/issues/1

### Description (What does it do?)
This is the first PR for this library. First of all, it adds basic boilerplate linting / formatting / github actions CI config. The main purpose of this PR is to get started setting up the primary purpose of the library, which is to generate a login button pointing at an instance of `mit-open`. The library exports a function called `initLoginButton`, which generates a login button pointing at a provided `mit-open` instance and a simple text element showing who is logged in if you already are logged in.

### Screenshots (if appropriate):
![image](https://github.com/mitodl/mit-open-login-button/assets/12089658/62c332e9-1b29-4603-80e6-5549e201770d)
![image](https://github.com/mitodl/mit-open-login-button/assets/12089658/be9a6b32-34e6-4021-a612-32c4cb6e789f)


### How can this be tested?
 - In order to test this, you will need a running instance of `mit-open` (https://github.com/mitodl/mit-open) on the `cg/login-button-cors-settings` branch. Follow the instructions in the `mit-open` readme for initial setup, check out the branch and spin it up. You will need an account in your local instance for this to work.
 - You will also need `ocw-hugo-themes` (https://github.com/mitodl/ocw-hugo-themes) set up locally, running a course using `yarn start course`. Follow the readme there for basic setup instructions and make sure you can start up a course. After you have that working, check out the `cg/add-mit-open-login-button` branch. Assuming you have cloned this repo, run `yarn add /path/to/mit-open-login-button` to add your local copy of the code for this to your locally running `ocw-hugo-themes`. This is necessary because the package is not yet published to NPM.
 - Once you have all this running, make sure you are logged out of your local instance of `mit-open`
 - Back in your locally running OCW course site, you should see a Login button in the upper right. Click this button and log in to your locally running `mit-open`
 - After you have logged in, you will be sent to http://localhost:8063
 - Go back to your locally running OCW site at http://localhost:3000 and confirm that you see `Logged in as: username` with your username in place of `username`
 - In another tab, log out of your locally running `mit-open`
 - Verify that the login button reappears
